### DISCUSSION TITLE
How does one populate search results from a custom async data source?

### DISCUSSION THREAD

**Discussion by designosis** [fomantic/Fomantic-UI#2888] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)]
I'm using the following ...
```js
        $('#search').search({
            type: 'category',
            showNoResults: false,
            minCharacters: 3,
            searchDelay: 400,
            source: [],
            onSearchQuery: function( query ){
                $('#search .ui.icon.input').addClass('loading');
                $('#search').search('cancel query');
                MyAJAX(query);
            },
            onSelect(result) {
                  mySelectFunction(result);
            }
        });
```
... to hijack the lovely search UI and get results the way I need to. I can't use API

**MyAXAJ** creates the following array ...
```js
    $res = [
        {"category":"Cat1", "title":"Title 1", "myID":"123"},
        {"category":"Cat1", "title":"Title 2", "myID":"634"},
        {"category":"Cat2", "title":"Title 3", "myID":"284"}
    ];
```
... and then spits out a javascript call like this:
```js
    $('#search .ui.icon.input').removeClass('loading');

    // WHAT WILL WORK HERE? (This does not)
    $('#search').search('add results',
        $('#search').search('generate results', { results: $res })
    );
```

This last bit is my failed attempt at showing my results via the category template, using code I found here: https://stackoverflow.com/questions/38033330/semantic-ui-search-custom-data-source.

(ALSO an FYI: The API callbacks on https://fomantic-ui.com/modules/search.html are broken.)

A super hacky solution is to call `$('#search').search('destroy');` then re-instantiate the `$('#search').search()`, passing `$res` as the source. But now I have to enter my search twice because the source wasn't there before. Which makes me thing there may be a way to trigger the search via `set value`, but that becomes so convoluted, and there must be a simpler way. I just need a way to push my custom results to Fomantic UI's search results, such that my `onSelect` function is passed the entire `result` (including `myID`)

Any advice on how to do this?

**Comment by designosis** [comment]
I'm curious, is this an extremely odd request?
Is it just not possible to populate search results without using the Fomantic API without massive coding chops?

**Reply by lubber-de** [reply]
Try `parse response` instead which will do `generateresults` and `addresults` and a few extra internal stuff

Havent tried myself though...
```javascript
$('#search').search('parse response', { results: $res });
```
